caas/kubernetes/provider/resources/validatingwebhookconfig.go

```diff
@@ -0,0 +1,131 @@
+
+	in.SetName(name)
+	return &ValidatingWebhookConfiguration{
+		client,
+		*in,
+	}
+}
+
+// Clone returns a copy of the resource.
+func (v *ValidatingWebhookConfiguration) Clone() Resource {
+	clone := *v
+	return &clone
+}
+
+// ID returns a comparable ID for the Resource
+func (v *ValidatingWebhookConfiguration) ID() ID {
+	return ID{"ValidatingWebhookConfiguration", v.Name, v.Namespace}
+}
+
+// Apply patches the resource change.
+func (v *ValidatingWebhookConfiguration) Apply(ctx context.Context) (err error) {
+	// attempt to create first, then patch if it already exists
+	created, err := v.client.Create(ctx, &v.ValidatingWebhookConfiguration, metav1.CreateOptions{
+		FieldManager: JujuFieldManager,
+	})
+	if err == nil {
+		v.ValidatingWebhookConfiguration = *created
+		return nil
+	}
+	if !k8serrors.IsAlreadyExists(err) {
+		return errors.Annotatef(err, "creating ValidatingWebhookConfiguration %q", v.GetName())
+	}
+
+	data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, &v.ValidatingWebhookConfiguration)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	res, err := v.client.Patch(ctx, v.Name, types.StrategicMergePatchType, data, metav1.PatchOptions{
+		FieldManager: JujuFieldManager,
+	})
+
+	if k8serrors.IsConflict(err) {
+		return errors.Annotatef(errConflict, "ValidatingWebhookConfiguration %q", v.Name)
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	v.ValidatingWebhookConfiguration = *res
+	return nil
+}
+
+// Get refreshes the resource.
+func (v *ValidatingWebhookConfiguration) Get(ctx context.Context) error {
+	res, err := v.client.Get(ctx, v.Name, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		return errors.NotFoundf("mutating webhook configuration: %q", v.Name)
+	} else if err != nil {
+		return errors.Trace(err)
+	}
+	v.ValidatingWebhookConfiguration = *res
+	return nil
+}
+
+// Delete removes the resource.
+func (v *ValidatingWebhookConfiguration) Delete(ctx context.Context) error {
+	err := v.client.Delete(ctx, v.Name, metav1.DeleteOptions{
+		PropagationPolicy: k8sconstants.DefaultPropagationPolicy(),
+	})
+	return errors.Trace(err)
+}
+
+// ComputeStatus returns a juju status for the resource.
+func (v *ValidatingWebhookConfiguration) ComputeStatus(ctx context.Context, now time.Time) (string, status.Status, time.Time, error) {
+	if v.DeletionTimestamp != nil {
+		return "", status.Terminated, v.DeletionTimestamp.Time, nil
+	}
+	return "", status.Active, now, nil
+}
+
+// ListValidatingWebhookConfigs returns a list of mutating webhook configs.
+func ListValidatingWebhookConfigs(ctx context.Context, client admissionclient.ValidatingWebhookConfigurationInterface, opts metav1.ListOptions) ([]*ValidatingWebhookConfiguration, error) {
+	var items []*ValidatingWebhookConfiguration
+	for {
+		res, err := client.List(ctx, opts)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		for _, item := range res.Items {
+			items = append(items, NewValidatingWebhookConfig(client, item.Name, &item))
```

Comment:
We're inconsistent in existing code. eg for storage classes, pvcs, and pods we do
```
items = append(items, StorageClass{StorageClass: v})
```
We we do a small drive by fix?

Reply:
refactored all with thanks, and changed to value instead of pointer to standardise with existing secrets